server/src/main/java/org/apache/druid/discovery/ClientUtils.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.discovery;
+
+import org.apache.druid.java.util.common.StringUtils;
+import org.apache.druid.java.util.http.client.Request;
+
+import javax.annotation.Nullable;
+import java.net.URL;
+import java.util.Iterator;
+
+/**
+ * Utils class for shared client methods
+ */
+public class ClientUtils
+{
+  @Nullable
+  public static String pickOneHost(DruidNodeDiscovery druidNodeDiscovery)
+  {
+    Iterator<DiscoveryDruidNode> iter = druidNodeDiscovery.getAllNodes().iterator();
+    if (iter.hasNext()) {
+      DiscoveryDruidNode node = iter.next();
+      return StringUtils.format(
+          "%s://%s",
+          node.getDruidNode().getServiceScheme(),
+          node.getDruidNode().getHostAndPortToUse()
+      );
+    }
+    return null;
+  }
+
+  public static Request withUrl(Request old, URL url)
+  {
+    Request req = new Request(old.getMethod(), url);
+    req.addHeaderValues(old.getHeaders());
+    if (old.hasContent()) {
+      req.setContent(old.getContent());
```

Comment:
```suggestion
      req.setContent(old.getContent()).copy();
```
We should probably copy the content values instead of sharing them